### PR TITLE
Add support for pathSymbols in asset filenames

### PR DESCRIPTION
**Pull request by mjauvin**

### CODE REVIEW

**Review by bennothommo — APPROVED** [review]
Looks fine to me, as long as it caters for assets on cloud storage - I know @LukeTowers and @jaxwilko put some work in towards that, so they may need to have a look.

**Comment by mjauvin** [comment]
> Looks fine to me, as long as it caters for assets on cloud storage - I know @LukeTowers and @jaxwilko put some work in towards that, so they may need to have a look.

It makes no big changes, just expands pathSymbols in $fileName argument. Otherwise, the behavior is unchanged.

**Comment by bennothommo** [comment]
I agree, but I just want to be safe ;)

File: modules/system/traits/AssetMaker.php [hunk 1]
```diff
@@ -295,6 +295,11 @@ public function getAssetPath($fileName, $assetPath = null)
             $assetPath = $this->assetPath;
         }
 
+        $symbolizedPath = File::localToPublic(File::symbolizePath($fileName));
+        if ($symbolizedPath) {
+            $fileName = $symbolizedPath;
+        }
```
Comment (LukeTowers) [inline modules/system/traits/AssetMaker.php]:
```suggestion
        // Process absolute or symbolized paths
        $publicPath = File::localToPublic(File::symbolizePath($fileName));
        if ($publicPath) {
            $fileName = $publicPath;
        }
```

**Pull request merged**